rsyslog/init.sls

```diff
@@ -50,6 +51,33 @@ rsyslog_mmjsonparse:
     - require:
       - pkg: rsyslog
 
+{%- else %}
+rsyslog_updates:
+  pkgrepo.managed:
+    - name: deb http://ppa.launchpad.net/adiscon/v8-devel/ubuntu focal main
+    - dist: focal
+
+rsyslog:
+  pkg.installed:
+    - name: rsyslog
+    - version: 8.2008.0~20200811150537-0adiscon1focal
+    - require:
+      - remove_rsyslog_8_16_0
+  file.managed:
+    - name: {{ rsyslog.config }}
+    - template: jinja
+    - source: {{ rsyslog.custom_config_template }}
+    - context:
+        config: {{ rsyslog|json }}
+
+rsyslog_mmjsonparse:
+  pkg.installed:
+    - name: rsyslog-mmjsonparse
+    - version: 8.2008.0~20200811150537-0adiscon1focal
+    - require:
```

Comment:
add a require here for the ppa repo you add before this

Reply:
Oh yeah! I've seen us not do this in a few places now and I'm honestly surprised we don't run into ordering issues more often.